packages/editor/src/components/page-attributes/order.js

```diff
@@ -25,13 +25,11 @@ export const PageAttributesOrder = ( { onUpdateOrder, order = 0 } ) => {
 	};
 	const value = orderInput === null ? order : orderInput;
 	return (
-		<TextControl
-			className="editor-page-attributes__order"
-			type="number"
+		<NumberControl
 			label={ __( 'Order' ) }
 			value={ value }
 			onChange={ setUpdatedOrder }
-			size={ 6 }
+			labelPosition="side"
```

Comment:
Because NumberControl has an option for a side label, the label ends up being more centered than it does in trunk.

Reply:
Nice one!

Reply:
This is what I'm seeing. Does it need a FlexBlock wrapper again?

<img src="https://user-images.githubusercontent.com/555336/214838146-6283ede0-9344-4c64-ac88-ff2f36197a02.png" alt="Input field not stretching to edge" width="300">

Reply:
Oh, nice find! It's aligned in FireFox but I see this in Safari now. I'll update it and check a few browsers to double-check. 👍